tests/test_gradient_descent.c

```diff
@@ -0,0 +1,52 @@
+#include <math.h>
+#include <stdio.h>
+
+#include "../src/1chipml.h"
+
+#define TOL 2.0e-4
+#define ITMAX 10
+#define N 2
+
+static int isAlmostEqual(const gradient_real *value,
+                         const gradient_real *expected, int n,
+                         double tolerance) {
+
+  for (int i = 0; i < n; i++) {
+    if (fabs(value[i] - expected[i]) > tolerance) {
+      printf("Fail: Expected %f got %f\n", expected[i], value[i]);
+      return 0;
```

Comment:
Should return 1 on error and 0 on success

Reply:
If you really want to return a boolean you should include stdbool.h, but everybody seems to hate this header file and I don't know why (And at this point I'm too afraid to ask...)